### ISSUE TITLE
version 4.0 little fixes

### ISSUE THREAD

**Issue by vilkun** [kmammou/v-hacd#104]
In Version 4.0 VHACD I replaced X, Y, Z with X_INDEX, Y_INDEX, Z_INDEX because it was compile issues with my UE project, where I use their vector's X, Y, Z members.
[1.txt](https://github.com/kmammou/v-hacd/files/8488983/1.txt)

**Comment by vilkun** [comment]
also there several define X, Y, Z in a file

**Comment by vilkun** [comment]
![image](https://user-images.githubusercontent.com/95850076/163388421-5f625610-233a-46a8-a914-751d7f6c5e2b.png)

**Comment by vilkun** [comment]
The whole patch can be
[2.txt](https://github.com/kmammou/v-hacd/files/8489031/2.txt)

**Comment by jratcliff63367** [comment]
Yikes, that's some ugly ass code. Not sure how that got in there. It's not mine. I will see if I can find a cleaner way to fix this.

**Comment by jratcliff63367** [comment]
Ok, thanks for catching this. That was as a result of two things. First, moving all of the implementation code into the header file; which caused the conflicts you saw. The second was just a lot of copy/paste that happened when merging a ton of utility code into the single monolithic header.

I have changed the #define X/Y/Z to be #define X_INDEX/Y_INDEX/Z_INDEX and removed the duplicate defines.

All methods that were previously called 'X', 'Y', 'Z', are now called 'getX', 'getY', 'getZ'.

Not only does this fix the build errors that you saw, it just makes the code cleaner and easier to understand.

Thanks for catching this. The fixes have been pushed to master.

**Comment by jratcliff63367** [comment]
How do I apply this patch? Sorry, I'm not familiar enough with the git process to do this. Can you make this in the form of a pull request?

**Comment by vilkun** [comment]
https://github.com/kmammou/v-hacd/pull/109
I applied it on top of 'version4' branch:
git clone -b version4 https://github.com/kmammou/v-hacd.git
cd v-hacd (think you forgot this?)
git apply /home/vilkun/work/vhacd_build/1/vhacd.txt
Also you can try:
git apply --reject --whitespace=fix mychanges.patch 
(see https://stackoverflow.com/questions/4770177/git-patch-does-not-apply )

It's a first time, when I write to really third party developers :)

**Comment by jratcliff63367** [comment]
Yeah, you should not be using the 'version4' branch. That was just a staging branch as I was developing the new version. That branch is not current or up to date. The only branch you should be working on now is simply the 'master' branch. Sorry for any confusion here.
